tests/apps/stateapp/app.go

```diff
@@ -0,0 +1,226 @@
+
+	var output []daprState
+	for _, state := range states {
+		value, err := get(state.Key)
+
+		if err != nil {
+			return nil, err
+		}
+
+		log.Printf("Result for get request for key %s: %v", state.Key, value)
+		output = append(output, daprState {
+			Key: state.Key,
+			Value: value,
+		})
+	}
+
+	log.Printf("Result for get request for %d states: %v", len(states), output)
+	return output, nil
+}
+
+func delete(key string) error {
+	log.Printf("Processing delete request for %s.", key)
+	url := strings.Join([]string { stateURL, key }, "/")
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return fmt.Errorf("Could not create delete request for key %s in Dapr: %s", key, err.Error())
+	}
+
+	log.Printf("Deleting state for %s", url)
+	client := &http.Client{}
+	_, err = client.Do(req)
+	if err != nil {
+		return fmt.Errorf("Could not delete key %s in Dapr: %s", key, err.Error())
+	}
+
+	return nil
+}
+
+func deleteAll(states []daprState) error {
+	log.Printf("Processing delete request for %d states.", len(states))
+
+	for _, state := range states {
+		err := delete(state.Key)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// handles all APIs
+func handler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Processing request for %s", r.URL.RequestURI())
+
+	// Retrieve request body contents
+	var req requestResponse
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		log.Printf("Could not parse request body: %s", err.Error())
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(requestResponse {
+			Message: err.Error(),
+		})
+		return
+	}
+
+    var res = requestResponse {}
+	var uri = r.URL.RequestURI()
+	var states []daprState
+	var statusCode = http.StatusOK
+
+	res.StartTime = epoch()
+	
+	switch uri {
+	case "/test/save":
+		err = save(req.States)
+	case "/test/get":
+		states, err = getAll(req.States)
+		res.States = states
+	case "/test/delete":
+		err = deleteAll(req.States)
+	default:
+		err = fmt.Errorf("Invalid URI: %s", uri)
+		statusCode = http.StatusBadRequest
+		res.Message = err.Error()
+	}
+
+	if err != nil && statusCode == http.StatusOK {
+		statusCode = http.StatusInternalServerError
+		res.Message = err.Error()
+	}
+
+	res.EndTime = epoch()
+
+	if statusCode != http.StatusOK {
+		log.Printf("Error status code %v: %v", statusCode, res.Message)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(statusCode)
+	json.NewEncoder(w).Encode(res)
+}
+
+// epoch returns the current unix epoch timestamp
+func epoch() int {
+	return (int)(time.Now().UnixNano() / 1000000)
+}
+
+// appRouter initializes restful api router
+func appRouter() *mux.Router {
+	router := mux.NewRouter().StrictSlash(true)
+
+	router.HandleFunc("/", indexHandler).Methods("GET")
+	router.HandleFunc("/test/save", handler).Methods("POST")
+	router.HandleFunc("/test/get", handler).Methods("POST")
+	router.HandleFunc("/test/delete", handler).Methods("POST")
```

Comment:
all endpoints point it out to the same handler. then can we use router.HandleFunc("/tests/{test}", handler).Methods("POST") which is the same endpoint of hellodapr?

Reply:
done.